benchmarking/runs/run_concentrate_all_attributes.py

```diff
@@ -0,0 +1,111 @@
+import math
+
+from benchmarking.data.interfaces import MockStudentProviderSettings
+from benchmarking.data.simulated_data.mock_student_provider import (
+    MockStudentProvider,
+)
+from benchmarking.evaluations.metrics.average_gini_index_multi_attribute import (
+    AverageGiniIndexMultiAttribute,
+)
+from benchmarking.evaluations.metrics.num_requirements_satisfied import (
+    NumRequirementsSatisfied,
+)
+from benchmarking.evaluations.metrics.num_teams_meeting_requirements import (
+    NumTeamsMeetingRequirements,
+)
+from benchmarking.evaluations.scenarios.concentrate_all_attributes import (
+    ConcentrateAllAttributes,
+)
+from benchmarking.simulation.simulation import Simulation
+from models.enums import ScenarioAttribute, Gender
+
+
+def run_concentrate_all_attributes():
+    """
+    Goal: Run concentrate all attributes scenario, measure average gini index across all attributes
+    """
+
+    # Defining our changing x-values (in the graph sense)
+    class_sizes = [100, 150, 200, 250, 300]
+    num_trials = 10
+    ratio_of_female_students = 0.5
+    races = [
+        "African",
+        "European",
+        "East Asian",
+        "South Asian",
+        "South East Asian",
+        "First Nations or Indigenous",
+        "Hispanic or Latin American",
+        "Middle Eastern",
+        "Other",
+    ]
+
+    for class_size in class_sizes:
+        print("CLASS SIZE /", class_size)
+
+        number_of_teams = math.ceil(class_size / 5)
+
+        # set up either mock or real data
+        student_provider_settings = MockStudentProviderSettings(
+            number_of_students=class_size,
+            number_of_friends=4,
```

Comment:
specifying friends and enemies here might also be making the social algorithm perform better.

approaching things in the scientific way here is good, where we want to ideally isolate the thing we want to observe as much as possible and in this case, giving people friends and enemies is not really part of what you explained this run was trying to find.

Or maybe you do specifically care about the case where students have these kinds of social relationships, but if so, we'd need to mention that in the comment of this run

Reply:
Good catch, I'll remove it cause it does make a big difference on the social algorithm. For 100 students, 0.71 with friends, to 0.46 without friends